src/main/java/seedu/duke/Storage.java

```diff
@@ -1,14 +1,196 @@
 package seedu.duke;
 
+import seedu.duke.command.Command;
+import seedu.duke.common.DateFormats;
 import seedu.duke.data.TransactionList;
+import seedu.duke.data.transaction.Transaction;
+import seedu.duke.exception.MoolahException;
+import seedu.duke.exception.StorageInputCorruptedException;
+import seedu.duke.exception.StorageWriteErrorException;
+import seedu.duke.parser.ParameterParser;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import static seedu.duke.parser.CommandParser.getCommand;
 
 public class Storage {
     //@@author chinhan99
+    private static final String DIRECTORY_PATH = "data";
+    private static final String FILE_PATH = "data/duke.txt";
+    private static final String DELIMITER = " \\| ";
+
+    private TransactionList storedTransactions;
 
     public Storage() {
+        this.storedTransactions = new TransactionList();
+    }
+
+
+    /**
+     * Checks if duke.txt and the directory exists. If any of these does not exist, it would be created.
+     *
+     * @return The duke.txt file which existed / is created.
+     * @throws IOException When there are issues creating the file.
+     */
+
+    private static File checkIfFileExist() throws IOException {
+        File directory = new File(DIRECTORY_PATH);
+        File file = new File(FILE_PATH);
+
 
+        if (!directory.exists()) {
+            directory.mkdir();
+            System.out.println("* Created new directory *");
+        }
+        if (!file.exists()) {
+            file.createNewFile();
+            System.out.println("* Created new file for use *");
+        } else {
+            System.out.println("* Existing file detected *");
+
+        }
+        return file;
     }
 
-    public void writeToFile(TransactionList transactions) {
+    /**
+     * Initializes the duke.txt by checking its existence, then store the data values to the program.
+     *
+     * @return The TransactionList storing entries which would be used in the program.
+     * @throws StorageInputCorruptedException If there are errors due to corrupted duke.txt data.
+     * @throws StorageWriteErrorException     If the file could not be created or could not be written.
+     */
+    public TransactionList initializeFile() throws MoolahException {
+        try {
+            File file = checkIfFileExist();
+            Scanner input = new Scanner(file);
+            storeFileValuesLocally(storedTransactions, input);
+            System.out.println("* duke.txt loaded successfully! *");
+
+        } catch (MoolahException e) {
+            //Catch any parsing errors and throw the default StorageInputCorruptedException
```

Comment:
There needs to be a single space between `//` and `Catch` as part of compliance with Javadoc commenting stye